client/src/legacy/entwine/BootstrapReactComponent.js

```diff
@@ -18,7 +18,38 @@ Injector.ready(() => {
           : {};
 
         const componentName = this.data('component');
-        const Component = loadComponent(componentName, context);
+
+        /** 
+         * Define variable "Component" after Injector found Component
+         * in existing Components
+        **/
+        let Component;
+
+        /**
+         * If Component name is set in data-component attribute and it is not empty string.
+         * Then try to get Component from existing Components.
+         * If Component does not exist load ErrorComponent.
+         * If Component exists load Component.
+         * If data-component attribute doesn't exist load DefaultComponent
+        **/
+        if (typeof componentName === 'string' && componentName !== '') {
+          let isExist = false;
+          //try to get Component from existing Components
+          try {
+            Injector.component.get(componentName);
+            isExist = true;
+          } catch (error) {
+            console.warn(error.message);
+          }
+          // if Component exists load Component
+          if (isExist) {
+            Component = loadComponent(componentName, context);
+          } else {
+            Component = loadComponent("ErrorComponent");
+          }
+        } else {
+          Component = loadComponent("DefaultComponent");
+        }
```

Comment:
That logic  will need to be adapted and moved to [ReactAdminBoostrap](https://github.com/maxime-rainville/silverstripe-react/blob/feature/react-admin/client/src/components/ReactAdminBoostrap.js)